---
title: 'Analyzing SARS-CoV-2 variants among the Spanish Islands and mainland'
author: "Alexandra Palacios"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable_Spanish_Islands_Madrid.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview


Here, I analyzed SARS-CoV-2 variants in regions within the different Spanish Islands and compared them with variants found in the Spanish capital, Madrid. 


This is a report on SARS-CoV-2, including some variant analysis [@koyama2020variant].

# Methods

## Data Collection

I selected and downloaded a total of 125 SARS-CoV-2 samples from the PRJEB43166 SRA Bioproject located in the NCBI SRA SARS-CoV-2 Bioproject list. This bioproject collected SARS-CoV-2 variant data from individuals of different ages, sex, and geographic region within Spain. 100 of the samples I selected came from localities within the Spanish Islands and were collected by Servicio de Microbiología, Hospital Universitario Son Espases and SeqCOVID-Spain consortium. The other 25 samples came from the Spanish capital, Madrid, and were collected by Hospital General Universitario Gregorio Marañón and SeqCOVID-Spain consortium.

## Variant Analysis

Using a bash pipeline created by [@koyama2020variant] and modified by Naupaka Zimmerman, I then downloaded all the raw Ilumina fastq data selected from the SRA Bioproject, checked the data quality using the , trimmed unwanted sequence data, indexed and mapped sequences against the SARS-CoV-2 reference genome, sorted and processed reads, and configured reads to be processed in R as vcf files. The SARS-CoV-2 reference genome came from the NCBI This entire pipeline was driven by a Makefile.

Next, I loaded in, tidied, and stacked all of the vcf files in R, loaded in a gff file with genome annotations

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

# Results

# Discussion

# Figures

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

```{r sex-figure}
# create bar plot that organizes samples by male and female
vcf_with_metadata %>%
  group_by(sample, host_sex) %>%
  tally() %>%
    ggplot(aes(x = host_sex)) +
    geom_bar() +
  labs(title = "Count of samples by sex", x = "Sex", y = "Count") +
  theme_few()
```

```{r age-figure}
# create bar plot that organizes samples by age group
vcf_with_metadata %>%
  group_by(sample, Host_Age) %>%
  tally() %>%
    ggplot(aes(x = Host_Age)) +
    geom_histogram(bins = 25) +
  labs(title = "Histogram of samples by host age", x = "Host Age", y = "Frequency") +
  theme_few()
```

``` {r age-by-sex-figure}
vcf_with_metadata %>%
  group_by(sample, Host_Age, host_sex) %>%
  tally() %>%
  ggplot(aes(x = Host_Age, fill = host_sex)) +
  geom_histogram() +
  facet_wrap(~ host_sex) +
  theme_few()
```

``` {r position-figure}
vcf_with_metadata %>%
  group_by(sample, pos, host_sex) %>%
  tally() %>%
  ggplot(aes(x = pos)) +
  geom_histogram() +
  theme_few()
```

```{r region-figure}
# create bar plot that organizes samples by Spanish region
vcf_with_metadata %>%
  mutate(`geographic_location_(region_and_locality)` = gsub("_", " ", `geographic_location_(region_and_locality)`)) %>%
  group_by(sample, `geographic_location_(region_and_locality)`) %>%
  tally() %>%
    ggplot(aes(x = `geographic_location_(region_and_locality)`)) +
    geom_bar() +
  labs(title = "Count of samples by geographic region", x = "Region", y = "Count") +
  theme(axis.text.x = element_text(angle = 90),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA))
```

```{r snp-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the grey background
```

**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

# Tables

``` {r spike-variant-table}
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(ref, alt, pos, gene) %>%
  filter(gene == "S") %>%
  tally() %>%
  select(ref, alt, pos, gene, n) %>%
  knitr::kable(col.names = c("Reference",
                             "Alternate",
                             "Position",
                             "Gene",
                             "Count"),
               align = "c")
```

**Table X**: Counts of individual variants for the spike protein from all 125 samples.

``` {r spike-variant-region-table}
vcf_with_metadata %>%
  filter(!is.na(gene)) %>%
  group_by(ref, alt, pos, gene, `geographic_location_(region_and_locality)`) %>%
  mutate(`geographic_location_(region_and_locality)` = gsub("_", " ", `geographic_location_(region_and_locality)`)) %>%
  filter(`geographic_location_(region_and_locality)` == "Madrid") %>%
  tally() %>%
  select(ref, alt, pos, gene, `geographic_location_(region_and_locality)`, n) %>%
  knitr::kable(col.names = c("Reference",
                             "Alternate",
                             "Position",
                             "Gene",
                             "Region",
                             "Count"),
               align = "c")
```

```{r gene-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 2**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
